services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/PortPropertiesSpec.groovy

```diff
@@ -0,0 +1,250 @@
+        northboundV2.updatePortProperties(sw.dpId, islToManipulate.srcPort,
+                new PortPropertiesDto(discoveryEnabled: true))
+        northboundV2.updatePortProperties(islToManipulate.dstSwitch.dpId, islToManipulate.dstPort,
+                new PortPropertiesDto(discoveryEnabled: true))
+
+        then: "System discovers link and link status is DISCOVERED"
+        Wrappers.wait(discoveryInterval + WAIT_OFFSET) {
+            assert islUtils.getIslInfo(islToManipulate).get().state == IslChangeType.DISCOVERED
+        }
+        Boolean testIsCompleted = true
+
+        cleanup: "Restore discovery port property on the port"
+        if (!testIsCompleted) {
+            northboundV2.updatePortProperties(islToManipulate.srcSwitch.dpId, islToManipulate.srcPort,
+                    new PortPropertiesDto(discoveryEnabled: true))
+            northboundV2.updatePortProperties(islToManipulate.dstSwitch.dpId, islToManipulate.dstPort,
+                    new PortPropertiesDto(discoveryEnabled: true))
+        }
+    }
+
+    def "Link is stop being discovered afrter disabling port discovery feature"() {
+        given: "A link with a flow on it"
+        def switchPair = getTopologyHelper().getAllNeighboringSwitchPairs().find { it.paths.size() >= 2 }
+        def flow = flowHelper.randomFlow(switchPair)
+        flowHelper.addFlow(flow)
+        def flowPath = northbound.getFlowPath(flow.id)
+        def islToManipulate = pathHelper.getInvolvedIsls(flowPath).first()
+
+        when: "Disable discovery port feature"
+        northboundV2.updatePortProperties(islToManipulate.srcSwitch.dpId, islToManipulate.srcPort,
+                new PortPropertiesDto(discoveryEnabled: false))
+
+        then: "Link state is changed to FAILED"
+        Wrappers.wait(discoveryTimeout + WAIT_OFFSET) {
+            assert islUtils.getIslInfo(islToManipulate).get().state == IslChangeType.FAILED
+        }
+
+        and: "Flow is rerouted"
+        def newFlowPath
+        Wrappers.wait(rerouteDelay + WAIT_OFFSET) {
+            newFlowPath = northbound.getFlowPath(flow.id)
+            newFlowPath != flowPath
+        }
+
+        when: "Enable discovery port property"
+        northboundV2.updatePortProperties(islToManipulate.srcSwitch.dpId, islToManipulate.srcPort,
+                new PortPropertiesDto(discoveryEnabled: true))
+
+        then: "Link state is changed to DISCOVERED"
+        Wrappers.wait(discoveryInterval + WAIT_OFFSET) {
+            assert islUtils.getIslInfo(islToManipulate).get().state == IslChangeType.DISCOVERED
+        }
+        Boolean testIsCompleted = true
+
+        and: "Flow is not rerouted"
+        northbound.getFlowPath(flow.id) == newFlowPath
+
+        cleanup: "Restore discovery port property on the port"
+        if (!testIsCompleted) {
+            northboundV2.updatePortProperties(islToManipulate.srcSwitch.dpId, islToManipulate.srcPort,
+                    new PortPropertiesDto(discoveryEnabled: true))
+        }
```

Comment:
Again, this is not how cleanup should look like. Flow is not removed, `testIsCompleted` usage is not valid